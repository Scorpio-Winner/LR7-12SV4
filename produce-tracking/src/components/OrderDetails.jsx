import React, { useState } from "react";
import { Box, Typography, Button } from "@mui/material";

const OrderDetails = ({ order, orderStatuses, companyInfo }) => {
  const [customerInfoOpen, setCustomerInfoOpen] = useState(false);

  const toggleCustomerInfo = () => {
    setCustomerInfoOpen(!customerInfoOpen);
  };

  // Find the corresponding customer information from companyInfo array
  const customer = companyInfo.find((company) => company.name === order.customer);

  return (
    <Box sx={{ display: 'flex', flexDirection: 'column', gap: '30px' }}>
      <Typography variant="h2" sx={{ alignSelf: 'center', fontSize: '26px' }}>
        {order.title}
      </Typography>
      <Box sx={{ display: 'flex', flexDirection: 'column', gap: '10px' }}>
        <Typography variant="body1" sx={{ width: '450px', fontSize: '18px' }}>
          <b>Id:</b> {order.id}
        </Typography>
        <Typography variant="body1" sx={{ width: '450px', fontSize: '18px' }}>
          <b>Status:</b> {order.status}
        </Typography>
        <Typography variant="body1" sx={{ width: '450px', fontSize: '18px' }}>
          <b>Creation date:</b> {order.creationDate}
        </Typography>
        <Typography variant="body1" sx={{ width: '450px', fontSize: '18px' }}>
          <b>Complete time:</b> {order.completeTime}
        </Typography>
        <Typography variant="body1" sx={{ width: '450px', fontSize: '18px' }}>
          <b>Description:</b><br />
          {order.description}
        </Typography>
        <Typography variant="body1" sx={{ width: '450px', fontSize: '18px' }}>
          <b>Customer:</b> {order.customer}
        </Typography>
        <Button variant="outlined" onClick={toggleCustomerInfo}>
          {customerInfoOpen ? "Hide Customer Details" : "Show Customer Details"}
        </Button>
        {customerInfoOpen && customer && (
          <Box>
            <Typography variant="body1" sx={{ width: '450px', fontSize: '18px' }}>
              <b>Location:</b> {customer.location}
            </Typography>
            <Typography variant="body1" sx={{ width: '450px', fontSize: '18px' }}>
              <b>Established Year:</b> {customer.establishedYear}
            </Typography>
            <Typography variant="body1" sx={{ width: '450px', fontSize: '18px' }}>
              <b>Industry:</b> {customer.industry}
            </Typography>
            <Typography variant="body1" sx={{ width: '450px', fontSize: '18px' }}>
              <b>Employees:</b> {customer.employees}
            </Typography>
            <Typography variant="body1" sx={{ width: '450px', fontSize: '18px' }}>
              <b>Revenue:</b> {customer.revenue}
            </Typography>
            <Typography variant="body1" sx={{ width: '450px', fontSize: '18px' }}>
              <b>Website:</b> {customer.website}
            </Typography>
            <Typography variant="body1" sx={{ width: '450px', fontSize: '18px' }}>
              <b>Description:</b> {customer.description}
            </Typography>
          </Box>
        )}
      </Box>
    </Box>
  );
};

export default OrderDetails;