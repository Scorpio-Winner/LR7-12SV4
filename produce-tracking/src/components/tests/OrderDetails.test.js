import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import OrderDetails from '../OrderDetails';

describe('OrderDetails', () => {
  const mockOrder = {
    id: '1',
    title: 'Test Order',
    status: 'Pending',
  };

  const mockCompanyInfo = {
    items: [
      {
        name: 'Test Order',
        location: 'Test Location',
        established_year: '2000',
        industry: 'Test Industry',
        employees: '100',
        revenue: '1000000',
        website: 'www.test.com',
        description: 'Test Description',
      },
    ],
  };

  it('renders correctly', () => {
    const { getByText } = render(
      <Router>
        <OrderDetails order={mockOrder} companyInfo={mockCompanyInfo} />
      </Router>
    );

    expect(getByText('Test Order')).toBeTruthy();
    expect(getByText('Pending')).toBeTruthy();
  });

  beforeEach(() => {
    localStorage.setItem('role', 'admin');
  });
  
  it('shows customer details when button is clicked', () => {
    const { getByText, queryByText } = render(
      <Router>
        <OrderDetails order={mockOrder} companyInfo={mockCompanyInfo} />
      </Router>
    );
  
    fireEvent.click(getByText('Show Customer Details'));
    expect(queryByText('Hide Customer Details')).toBeTruthy();
  });
  

  it('hides customer details when button is clicked twice', () => {
    const { getByText, queryByText } = render(
      <Router>
        <OrderDetails order={mockOrder} companyInfo={mockCompanyInfo} />
      </Router>
    );

    fireEvent.click(getByText('Show Customer Details'));
    fireEvent.click(getByText('Hide Customer Details'));
    expect(queryByText('Location: Test Location')).toBeNull();
  });

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <OrderDetails order={mockOrder} companyInfo={mockCompanyInfo} />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches the snapshot when customer details are shown', () => {
    const { getByText } = render(
      <Router>
        <OrderDetails order={mockOrder} companyInfo={mockCompanyInfo} />
      </Router>
    );

    fireEvent.click(getByText('Show Customer Details'));

    const tree = renderer
      .create(
        <Router>
          <OrderDetails order={mockOrder} companyInfo={mockCompanyInfo} />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
