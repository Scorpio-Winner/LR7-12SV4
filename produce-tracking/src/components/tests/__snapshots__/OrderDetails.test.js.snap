// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDetails matches the snapshot 1`] = `
<div
  className="MuiBox-root css-4p9vrj"
>
  <h2
    className="MuiTypography-root MuiTypography-h2 css-acoteq-MuiTypography-root"
  >
    Test Order
  </h2>
  <div
    className="MuiBox-root css-bco1gb"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-11nzmzd-MuiTypography-root"
    >
      <b>
        Id:
      </b>
       
      1
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-11nzmzd-MuiTypography-root"
    >
      <b>
        Status:
      </b>
       
      Pending
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-11nzmzd-MuiTypography-root"
    >
      <b>
        Creation date:
      </b>
       
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-11nzmzd-MuiTypography-root"
    >
      <b>
        Complete time:
      </b>
       
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-11nzmzd-MuiTypography-root"
    >
      <b>
        Description:
      </b>
      <br />
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-11nzmzd-MuiTypography-root"
    >
      <b>
        Customer:
      </b>
       
    </p>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Show Customer Details
    </button>
  </div>
</div>
`;

exports[`OrderDetails matches the snapshot when customer details are shown 1`] = `
<div
  className="MuiBox-root css-4p9vrj"
>
  <h2
    className="MuiTypography-root MuiTypography-h2 css-acoteq-MuiTypography-root"
  >
    Test Order
  </h2>
  <div
    className="MuiBox-root css-bco1gb"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-11nzmzd-MuiTypography-root"
    >
      <b>
        Id:
      </b>
       
      1
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-11nzmzd-MuiTypography-root"
    >
      <b>
        Status:
      </b>
       
      Pending
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-11nzmzd-MuiTypography-root"
    >
      <b>
        Creation date:
      </b>
       
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-11nzmzd-MuiTypography-root"
    >
      <b>
        Complete time:
      </b>
       
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-11nzmzd-MuiTypography-root"
    >
      <b>
        Description:
      </b>
      <br />
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-11nzmzd-MuiTypography-root"
    >
      <b>
        Customer:
      </b>
       
    </p>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Show Customer Details
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
